client.go

```diff
@@ -41,6 +41,11 @@ func New(address string) (*Client, error) {
 	return NewForUser(address, username)
 }
 
+// Return a New Client with auto config
+func AutoConfigClient() (*Client, error) {
```

Comment:
Rather than a new method, let's just have this be default fallback behavior for `New` if an empty string is passed for the namenode.